### PR TITLE
Clean tests -- touch up test suite

### PR DESCRIPTION
**Pull request by ztatlock**
@d1jang @Ptival can you check out this clean up pass please?

### CODE REVIEW

File: compiler/src/kernel.ml [hunk 1]
```diff
@@ -102,7 +102,7 @@ let empty_kernel =
   ; components = []
   ; msg_decls  = []
   ; init       = Nop
-  ; exchange   = ("", [])
+  ; exchange   = ("c", [])
```
Comment (d1jang) [inline compiler/src/kernel.ml:105]: I don't understand what this change is about. what does "c" stand for??

File: compiler/test/echo-00-c/test.sh [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+  GDB=gdb
+fi
+
 if [ "$KRAKEN_GDB" != "" ]; then
   echo "
   break main
   run $@
   " > $KROOT/client/gdb-run
-  cgdb $KROOT/client/test -x $KROOT/client/gdb-run
+  $GDB $KROOT/client/test -x $KROOT/client/gdb-run
```
Comment (d1jang) [inline compiler/test/echo-00-c/test.sh:12]: are we using GDB instead of ocaml debug??

File: compiler/test/echo-00/echo-auto-00/kernel.krn [hunk 3]
```diff
@@ -23,14 +20,18 @@ Exchange(c) {
...
 Properties {
   AlwaysResponds =
-    ImmAfter
-      { "KRecv chan msg :: nil" }
-      { "KSend chan msg :: nil" };
+    ImmAfter {
+      "KRecv chan msg :: nil"
+    }{
+      "KSend chan msg :: nil"
+    };
```
Comment (d1jang) [inline compiler/test/echo-00/echo-auto-00/kernel.krn:28]: What's ImmAfter about?

File: compiler/test/echo-00/echo-auto-00/kernel.krn [hunk 4]
```diff
@@ -23,14 +20,18 @@ Exchange(c) {
...
 
   OnlyResponds =
-    ImmBefore
-      { "KRecv chan msg :: nil" }
-      { "KSend chan msg :: nil" };
+    ImmBefore {
+      "KRecv chan msg :: nil"
+    }{
+      "KSend chan msg :: nil"
+    };
```
Comment (d1jang) [inline compiler/test/echo-00/echo-auto-00/kernel.krn:35]: What's ImmBefore about?

File: compiler/src/kernel.ml (hunk 1 above)
Comment (ztatlock) [inline compiler/src/kernel.ml:105]: Good question @d1jang 

This is the "empty_kernel" which we fill in during parsing.

In the case that the Kraken spec we are compiling doesn't have an Exchange section, we still want to generate code that compiles.

To support that, the default Exchange needs to have /some/ channel to receive messages on.

Thus "c" here is just a default, dummy channel name.

File: compiler/test/echo-00-c/test.sh (hunk 2 above)
Comment (ztatlock) [inline compiler/test/echo-00-c/test.sh:12]: This is for running a client written in C, so we have to use gdb when we want to debug it.

File: compiler/test/echo-00/echo-auto-00/kernel.krn (hunk 3 above)
Comment (ztatlock) [inline compiler/test/echo-00/echo-auto-00/kernel.krn:28]: ImmAfter is a property to prove on trace fragments occuring in the traces of KInvariant states.

From KProp.v :

``` ocaml

(* if you see bef in a trace,
 * then it is immediately chrono followed by aft
 *)
Definition ImmAfter (bef aft : KTrace) : Prop :=
  forall kst, KInvariant kst ->
  forall x y, ktr kst = [x ++ bef ++ y]%inhabited ->
  exists z, x = z ++ aft.


```

File: compiler/test/echo-00/echo-auto-00/kernel.krn (hunk 4 above)
Comment (ztatlock) [inline compiler/test/echo-00/echo-auto-00/kernel.krn:35]: It's like ImmAfter but with the order flipped:

``` ocaml
(* if you see aft in a trace,
 * then it is immediately chrono preceded by bef
 *)
Definition ImmBefore (bef aft : KTrace) : Prop :=
  forall kst, KInvariant kst ->
  forall x y, ktr kst = [x ++ aft ++ y]%inhabited ->
  exists z, y = bef ++ z.

```

**Comment by d1jang** [comment]
Everything seems ready to be merged!

**Pull request merged**